gcp_variant_transforms/beam_io/vcf_estimate_io.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2019 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""A source for reading VCF files and extracting signals about input size."""
+
+from __future__ import absolute_import
+
+from functools import partial
+from typing import Dict, Iterable  # pylint: disable=unused-import
+
+from apache_beam import transforms
+from apache_beam.io import filebasedsource
+from apache_beam.io import filesystem
+from apache_beam.io import filesystems
+from apache_beam.io import iobase
+from apache_beam.io import range_trackers  # pylint: disable=unused-import
+
+
+class VcfEstimate(object):
+  """Container for estimation data about the VCF file."""
+
+  def __init__(self,
+               file_name,  # type: str
+               estimated_variant_count,  # type: float
+               samples,  # type: List[str]
+               size_in_bytes  # type: int
+              ):
+    # type: (...) -> None
+    """Initializes a VcfEstimate object.
+
+    Args:
+      file_name: Name of file.
+      estimated_variant_count: Estimated number of variants.
+      samples: Sample names in the file.
+      size_in_bytes: Size of the file.
+    """
+    self.file_name = file_name
+    self.estimated_variant_count = estimated_variant_count
+    self.samples = samples
+    self.size_in_bytes = size_in_bytes
+
+  def __eq__(self, other):
+    return self.file_name == other.file_name
+
+  def __repr__(self):
+    return 'File Name: {}, Variant Count: {}, Samples: {}, Size: {}'.format(
+        self.file_name,
+        self.estimated_variant_count,
+        self.samples,
+        self.size_in_bytes
+    )
+
+
+class VcfEstimateSource(filebasedsource.FileBasedSource):
+  """A source for reading VCF file estimates."""
```

Comment:
"""A source that produces estimates about the number of variants and samples in a VCF file.""" ?

This makes it sound like it reads estimates